modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/introspection/validation/NameClashModelValidator.java

```diff
@@ -66,13 +75,15 @@ public void validate(ExtensionModel model) throws IllegalModelDefinitionExceptio
 
     private final ExtensionModel extensionModel;
     private final Set<DescribedReference<Named>> namedObjects = new HashSet<>();
+    private final Map<String, DescribedParameter> singularizedObjects = new HashMap<>();
     private final Multimap<String, TopLevelParameter> topLevelParameters = LinkedListMultimap.create();
 
     public ValidationDelegate(ExtensionModel extensionModel) {
       this.extensionModel = extensionModel;
     }
 
     private void validate(ExtensionModel extensionModel) throws IllegalModelDefinitionException {
```

Comment:
The actual logic doesn't check clashing between `@Configuration( name= "SomeConfig") ` and `@Configuration( name= "some-config")`